assigner/__init__.py

```diff
@@ -186,6 +187,8 @@ def main(args=sys.argv[1:]):
             raise e
```

Comment:
I don't like this anymore (I can't comment on the `except Exception as e:`). We should do special `except KeyError as e:` or `except GitCommandNotFound as e:` statements for each case. Then, once we run out, catch `Exception` and log them. As a `finally:` clause we should do tracebacks / `SystemExit` raises.

Reply:
What do YOU think?

Reply:
Would `e` be in scope for a `finally:` block?

The reason this is structured as it is is lines 185-186 -- if we have tracebacks turned on, we don't want to do any special logging, just re-raise the exception. If we were to split this into several `except` clauses, we'd have to replicate that logic across each.

Reply:
Oh that's a good question. Probably not. You could get e but it'd be hacky at that point.